qcfractal/interface/models/gridoptimization.py

```diff
@@ -0,0 +1,208 @@
+"""
+A model for GridOptimization
+"""
+
+import json
+from typing import Any, Dict, List, Tuple, Union
+
+from pydantic import BaseModel, validator
+
+from .common_models import QCMeta, Provenance, Molecule, json_encoders, hash_dictionary
+
+__all__ = ["GridOptimizationInput", "GridOptimization"]
+
+
+class ScanDimension(BaseModel):
+    """
+    A dimension to scan over
+    """
+    type: str
+    indices: List[int]
+    steps: List[float]
+
+    @validator('type')
+    def check_type(cls, v, values, **kwargs):
+        possibilities = {"distance", "angle", "dihedral"}
+        if v not in possibilities:
+            raise TypeError("Type '{}' found, can only be one of {}.".format(v, possibilities))
+
+        return v
+
+    @validator('indices', whole=True)
+    def check_indices(cls, v, values, **kwargs):
```

Comment:
That seems like an interesting design choice which makes the `values` in general a bit dangerous. Looking through the source code, I do not see anything that would trigger this. Can you provide an example that has this problem?

Reply:
The example here provides a case where one field will have access to the other, but not vice versa

https://pydantic-docs.helpmanual.io/#validators

Reply:
I see so values is a mirror of the objects dict. I think this is the other way around from what you mentioned where the validators are called in order and `values` updates itself only if the validation passes. Since the order is static this should be quite safe.